app/controllers/users.js

```diff
@@ -44,6 +44,98 @@ exports.findUser = (req, res) => {
   });
 };
 
+exports.getFriends = (req, res) => {
+  const userId = req.body.userId;
+  User.findById(userId, (err, user) => {
+    if (!err) {
+      res.send(user.friends);
+    } else {
+      res.send('An error occurred');
+    }
+  });
+};
+
+exports.addFriends = (req, res) => {
+  const userToAddEmail = req.body.user;
+  const userId = req.body.userId;
+  User.findOne({
+    email: userToAddEmail
+  }, (err, obj) => {
+    if (!err) {
+      res.send(obj.name);
+    } else {
+      res.send('An error occurred');
+    }
+  });
+  User.findOneAndUpdate({
+    _id: userId
+  }, {
+    $push: {
+      friends: userToAddEmail
+    }
+  }, {
+    safe: true,
+    upsert: true
+  }, (error) => {
+    if (!(error)) {
+      return error;
+    }
+  });
+
+  User.findById(userId, (err, user) => {
+    if (err) throw err;
```

Comment:
Probably should respond with appropriate status codes here as well instead of throwing an error.